pkg/webhook/admission/pod/storage_initializer_injector.go

```diff
@@ -683,6 +689,42 @@ func (mi *StorageInitializerInjector) createModelContainer(image string, modelPa
 	return modelContainer
 }
 
+func (mi *StorageInitializerInjector) createModelInitContainer(image string) *v1.Container {
+	cpu := mi.config.CpuModelcar
+	if cpu == "" {
+		cpu = CpuModelcarDefault
+	}
+	memory := mi.config.MemoryModelcar
+	if memory == "" {
+		memory = MemoryModelcarDefault
+	}
```

Comment:
I'm unsure if you need to set those limits to the same value as the real model car. But it's ok as both the command for checking the conformance of the modelcar image and the actual command executed when running as a model car should be similar and minimal. It should also not affect the scheduling of the Pod, as for the scheduling decision, the max(resources of init-containers) is used, while for sidecars they are added up (as they run in parallel).

However, I would probably set both `requests` and `limits` to the same values to avoid the Pod not being put into the QoS class of "Guaranteed" (which is the highest possible class). Let's do this, too, for the modelcar when running as a sidecar (if not already)

See https://kubernetes.io/docs/tasks/configure-pod-container/quality-service-pod/ for more details.

Reply:
I think somewhat confused. [Some lines later](https://github.com/kserve/kserve/pull/3932/files#diff-19ab20f87e5f9bff37c2eada3e341c832cf91029b78a75b9f70acf6a22dacb8cR711), resources and limits are set. I see you have some comments around those lines :-) 

From [this docs](https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/#resource-requests-and-limits-of-pod-and-container), I'm not seeing any additional attributes to set, unless I'm not seeing well. So, I think I need you to elaborate a little bit more.

Reply:
Sorry, my bad :) Forget my comments, I just was confused.